### PR TITLE
refactor: replace references to ZClientViewController by abstractions - WPB-9801

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9801" title="WPB-9801" target="_blank">Sub-taskWPB-9801</a>  Refactoring
  </td></table>
  <br />
 

### Issue

The ZClientViewController is directly accessed from many places in the code.
As a starting point to fix this, this PR creates different protocols and wrappers, which are then injected.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios/Wire-iOS Tests/GroupParticipantsDetailViewControllerTests.swift [hunk 1]
```diff
@@ -99,11 +104,12 @@ final class GroupParticipantsDetailViewControllerTests: XCTestCase {
...
+        let createSut: () -> UIViewController = { [self] in
             let sut = GroupParticipantsDetailViewController(
                 selectedParticipants: selected,
                 conversation: conversation,
-                userSession: self.userSession
+                userSession: userSession,
+                mainCoordinator: mockMainCoordinator
             )
             return sut.wrapInNavigationController()
         }
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS Tests/GroupParticipantsDetailViewControllerTests.swift]: Can this become a helper function (that doesn't need to capture self)?

File: wire-ios/Wire-iOS/Sources/AppDelegate.swift [hunk 2]
```diff
@@ -84,6 +84,7 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
         return UIApplication.shared.delegate as! AppDelegate
     }
 
+    // TODO [WPB-9867]: remove this property
     var mediaPlaybackManager: MediaPlaybackManager? {
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/AppDelegate.swift:88]: 👍

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController.swift [hunk 3]
```diff
@@ -31,6 +31,8 @@ enum ConversationListState {
 final class ConversationListViewController: UIViewController {
 
     let viewModel: ViewModel
+    let mainCoordinator: MainCoordinating
+    weak var zClientViewController: ZClientViewController?
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController.swift:36]: `weak` is the right choice! 💪

File: wire-ios/Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift [hunk 4]
```diff
@@ -24,19 +24,55 @@ import WireSyncEngine
...
+
+    private var selfProfileViewControllerBuilder: SelfProfileViewControllerBuilder {
+        .init(
+            selfUser: userSession.editableSelfUser,
+            userRightInterfaceType: UserRight.self,
+            userSession: userSession,
+            accountSelector: SessionManager.shared
+        )
+    }
+    // TODO: make completely private
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift]: misses ticket reference

File: wire-ios/Wire-iOS/Sources/UserInterface/MainCoordinator/MainCoordinator.swift [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireDataModel
+import WireSystem
+
+struct MainCoordinator_: MainCoordinating {
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/MainCoordinator/MainCoordinator.swift]: Does it have the underscore suffix on purpose?

**Review by markusfassbender — APPROVED** [review]
Important improvement!

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/MainCoordinator/MainCoordinator.swift (hunk 5 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/UserInterface/MainCoordinator/MainCoordinator.swift]: No, I missed that, thank you.

File: wire-ios/Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift (hunk 4 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift]: fixed: fcd249b649dc3d9b5a8638658a4082a765b3c87b

**Pull request merged**